src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -247,7 +246,7 @@ public void Add(T item)
             var array = _items;
             var size = _size;
             _version++;
-            if ((uint)size < (uint)array.Length)
+            if (array.IndexInRange(size))
```

Comment:
> In the List`1.Add(long) dasm I was looking at, it never put the length in a register

Hmm, in other cases I looked at this seems to work as expected. Anyway, I think that using this trick in `List<T>.Add` is good now that it inlines. Everyone using it will now pay the cost of increased code size so it make sense to try to minimize that code considering that `Add` is rather common.